WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductDetailViewModel.kt

```diff
@@ -360,13 +360,22 @@ class ProductDetailViewModel @Inject constructor(
         updateProductBeforeEnteringFragment()
     }
 
+    /**
+     * Called during the Add _first_ Variation flow. Uploads the pending attribute changes and generates the first
+     * variation for the variable product.
+     */
     fun onAttributeListDoneButtonClicked() {
         saveAttributeChanges()
         attributeListViewState = attributeListViewState.copy(isCreatingVariationDialogShown = true)
         launch {
             viewState.productDraft?.let { draft ->
                 variationRepository.createEmptyVariation(draft)
-                    ?.let { updateProductDraft(numVariation = draft.numVariations + 1) }
+                    ?.let {
+                        val fetchedProduct = productRepository.fetchProduct(draft.remoteId)
+                        if (fetchedProduct != null) {
+                            updateProductState(productToUpdateFrom = fetchedProduct)
+                        }
+                    }
```

Comment:
np: I think we would benefit of using a pipe operation for each of these operation to keep nullability handling in check, WDYT? Something like this:
```suggestion
                    ?.let { productRepository.fetchProduct(draft.remoteId) }
                    ?.let { updateProductState(productToUpdateFrom = it) }
```

Reply:
@ThomazFB So I tested this and it looks like using this chain will result in different behavior. If `productRepository.fetchProduct` fails for some reason and returns `null`, then we would be triggering `ExitProductAttributeList` with **`variationsCreated` set to `false`**.

https://github.com/woocommerce/woocommerce-android/blob/98ae08c78b0bf0999290917dacbbb42e355a506d/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductDetailViewModel.kt#L380

And that will navigate the user _back_ instead of navigating them to the product detail page. 

I think this is not ideal and it's better to still navigate the user to the product detail page even if  `productRepository.fetchProduct` fails. What do you think? How can I present this logic using `?.let` or another magical chaining? 😄

Reply:
I see, in that case, the only improvement I see is that we could avoid the direct `!= null` comparison just for syntax cohesion and use the `also` function as the side result operation from the `fetchProduct` call. Something like this maybe:

```
viewState.productDraft?.let { draft ->
                variationRepository.createEmptyVariation(draft)
                    ?.let {
                        productRepository.fetchProduct(draft.remoteId)
                            ?.also { updateProductState(productToUpdateFrom = it) }
                    }
                    ?.let { triggerEvent(ExitProductAttributeList(variationCreated = true)) }
                    ?: triggerEvent(ExitProductAttributeList())
            }.also {
                attributeListViewState = attributeListViewState.copy(isCreatingVariationDialogShown = false)
            }
``` 

WDYT? I'm fine if you want to leave as it is too, I guess I'm being too nitpicky about this 😅

Reply:
Ooh cool. Applied in 40f60b441a8f2c185ef90684187e4dff400a5d17.

>  I guess I'm being too nitpicky about this

Not at all. I'm rusty so I appreciate getting introduced to new things.